packages/e2e-test-utils-playwright/src/request-utils/themes.ts

```diff
@@ -13,12 +13,25 @@ async function activateTheme(
 	let response = await this.request.get( THEMES_URL );
 	const html = await response.text();
 	const optionalFolder = '([a-z0-9-]+%2F)?';
-	const matchGroup = html.match(
-		`action=activate&amp;stylesheet=${ optionalFolder }${ encodeURIComponent(
+
+	// The `optionalFolder` regex part matches paths with a folder,
+	// so it will return the first match, which might contain a folder.
+	// First try to honor the including theme slug, that is, without a folder.
+	let matchGroup = html.match(
+		`action=activate&amp;stylesheet=${ encodeURIComponent(
 			themeSlug
 		) }&amp;_wpnonce=[a-z0-9]+`
 	);
 
+	// If the theme is not found, try to match the theme slug with a folder.
+	if ( ! matchGroup ) {
+		matchGroup = html.match(
+			`action=activate&amp;stylesheet=${ optionalFolder }${ encodeURIComponent(
```

Comment:
The second match is to honor the intentions of:

- https://github.com/WordPress/gutenberg/pull/48206